apiservice/apiservice.go

```diff
@@ -0,0 +1,240 @@
+package apiservice
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow/protobuf/go/flow/access"
+
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/grpcutils"
+)
+
+func NewFlowAPIService(accessNodeAddressAndPort flow.IdentityList, timeout time.Duration) (*FlowAPIService, error) {
+	accessClients := make([]access.AccessAPIClient, accessNodeAddressAndPort.Count())
+	for i, identity := range accessNodeAddressAndPort {
+		if identity.NetworkPubKey == nil {
+			clientRPCConnection, err := grpc.Dial(
+				identity.Address,
+				grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
+				grpc.WithInsecure(), //nolint:staticcheck
+				backend.WithClientUnaryInterceptor(timeout))
+			if err != nil {
+				return nil, err
+			}
+
+			accessClients[i] = access.NewAccessAPIClient(clientRPCConnection)
+		} else {
+			tlsConfig, err := grpcutils.DefaultClientTLSConfig(identity.NetworkPubKey)
```

Comment:
The `identity.NetworkPubKey` isn't the TLS public key used by the AN, the TLS keys are derived from the networking key.

This is how we setup secure connections elsewhere:
https://github.com/onflow/flow-go/blob/master/utils/grpcutils/grpc_client.go#L14-L31

Reply:
Looking...

Reply:
Addressed: https://github.com/GetElastech/flow-go/commit/368497cdbc5574eb5af0e10d4d54561222e8aa2a

Reply:
To be more precise it is first addressed here: https://github.com/GetElastech/flow-go/commit/58aa00f8776a9039272865af733cd5e60df5eb2c